container_test.go

```diff
@@ -506,3 +511,35 @@ func TestParseDockerIgnore(t *testing.T) {
 		assert.Equal(t, testCase.expectedExcluded, excluded)
 	}
 }
+
+func ExampleGenericContainer_withSubstitutors() {
+	ctx := context.Background()
+
+	// applyImageSubstitutors {
+	container, err := GenericContainer(ctx, GenericContainerRequest{
+		ContainerRequest: ContainerRequest{
+			Image:             "alpine:latest",
+			ImageSubstitutors: []ImageSubstitutor{dockerImageSubstitutor{}},
```

Comment:
I missed this in a previous PR, but wonder in which uses cases do we need more than one substitute. If a substitute is added then the new image name should be pull from the registry, in this case, what happens when more than one is added. If it is not intended then should we deprecate it in a next PR and only accept one?

Reply:
Mmmm I see your point, now with more perspective given the context from this PR. Yeah, makes sense. We just want one function plus the prefix prepender one, in the case it exists.